guide/index.adoc

```diff
@@ -0,0 +1,929 @@
+
+==== Log Everything
+Not having direct access to the file system of your job means the only method for feedback on what is happening inside the container is through console output. 
+Use standard output and/or standard error to indicate any progress or errors for your algorithm. 
+Some languages (such as Python) may require specifying that output should not be buffered until the process exits. 
+This will facilitate live viewing of output for long-running processes.
+
+==== Meaningful Errors
+Specific error conditions represented by unique process exit codes should be used to give clear feedback when a known error has occurred. 
+These can be used to distinguish between data errors where the data arrived but contained an error that could not be automatically addressed by the job and errors that occurred within the job execution itself. 
+See also <<Why should I define error codes?>> in the FAQ section.
+
+==== Runtime Configuration
+Most jobs have configuration values that are desirable to change at runtime. 
+These could be anything from confidence values for feature detection, to a band number within a multi-band image. 
+Ensure that your job can consume its configuration as either environment variables or command-line arguments. 
+This makes it possible to define your job as a configurable Docker image and avoid requiring the creation of new Docker images for common changes.
+
+==== Privilege Minimization
+Docker images often are set to use the `root` user by default. 
+Avoid this unless absolutely necessary. 
+Instead of using the `root` user, add a non-privileged user to the Docker image and set the {docker-run-user-url}[USER] for the image to this non-privileged user.
+
+==== Dockerfiles
+Docker provides guidance on some additional {docker-best-practices-url}[best practices] for writing Dockerfiles.
+
+
+=== Frequently Asked Questions (FAQs)
+
+==== What is this Seed image?
+{seed-url}[Seed] is a general {seed-url}/seed.html[standard] that was developed to aid in the discovery and running of self-contained algorithms. 
+It is a JSON-defined manifest that is embedded into the Docker image as a label. 
+It defines:
+
+	* Algorithm - name, version, description
+	* Developer - name, company, contact info
+	* Inputs/Outputs - how many, what type
+	* Hardware Requirements - CPUs, memory, disk
+	* Error handling - data vs. algorithm errors
+	* Environment variables
+	* Mounts
+
+==== How do I use Docker?
+Many resources for learning about Docker can be found on the https://www.docker.com[Docker website]. 
+The primary goal of this guide is to explain how to integrate an algorithm with Seed and not focus on how to build a Docker container.
+
+==== Why not just add all the files in my directory to the Docker image?
+The image should be as small as necessary. It is pulled across the network each time it runs on a node.  
```

Comment:
> ... as small as possible and only include the necessary files.

Reply:
Updated in [df17722](https://github.com/ngageoint/seed-cli/pull/233/commits/df1772214e5dfcaf9c9bf7d76daa0a62fda86b0c)